ReactCommon/hermes/inspector/chrome/Connection.cpp

```diff
@@ -1411,20 +1411,14 @@ Connection::Impl::makePropsFromValue(
 }
 
 void Connection::Impl::handle(const m::runtime::GetHeapUsageRequest &req) {
-  auto resp = std::make_shared<m::runtime::GetHeapUsageResponse>();
-  resp->id = req.id;
-
-  inspector_
-      ->executeIfEnabled(
-          "Runtime.getHeapUsage",
-          [this, req, resp](const debugger::ProgramState &state) {
-            auto heapInfo = getRuntime().instrumentation().getHeapInfo(false);
-            resp->usedSize = heapInfo["hermes_allocatedBytes"];
-            resp->totalSize = heapInfo["hermes_heapSize"];
-          })
-      .via(executor_.get())
-      .thenValue([this, resp](auto &&) { sendResponseToClient(*resp); })
-      .thenError<std::exception>(sendErrorToClient(req.id));
+  runInExecutor(req.id, [this, req]() {
+    auto heapInfo = getRuntime().instrumentation().getHeapInfo(false);
+    auto resp = std::make_shared<m::runtime::GetHeapUsageResponse>();
+    resp->id = req.id;
+    resp->usedSize = heapInfo["hermes_allocatedBytes"];
+    resp->totalSize = heapInfo["hermes_heapSize"];
+    sendResponseToClient(*resp);
+  });
 }
 
 void Connection::Impl::handle(const m::runtime::GetPropertiesRequest &req) {
```

Comment:
Should we update also this function?

Reply:
I dont this so, it acceses some fields that need to be accessed from the insepector thread.